cmd_search.c

```diff
@@ -142,39 +153,36 @@ int cmd_search(context_t *context) {
   consume_args(context, optind);
 
   /* We require a pattern or a pid to search for */
-  if (context->argc < 1 && search.pid == 0) {
+  if ((context->argc < 1 && search.pid == 0) &&
+      !(search.searchmask & (SEARCH_NAME | SEARCH_CLASS | SEARCH_CLASSNAME))) {
```

Comment:
Minor note: SEARCH_PID could also just be looked for in the searchmask here as in line 164.  Effect is the same, not sure which is clearer.